### PR TITLE
Enable linux integration tests

### LINKED ISSUE #1104: [Request]: Integration tests for Linux
### Plugin

all

### Use case

At the moment integration_test run on Android and iOS emulator/simulator but not on desktop

### Proposal

Enable the integration_test to run and pass on a Linux "ubuntu-latest" machine.

Since these machines images are "headless", i.e. there's no UI, not sure if Flutter integration tests can run on them, so this needs a bit of R&D.

Also, the tests will need to be adapted to Linux.

### PR DESCRIPTION
**Pull request by abd99**
## Description

Added CI workflow steps to run integration tests on Linux

- `battery_plus`
- `connectivity_plus`
- `device_info_plus`
- `network_info_plus`
- `package_info_plus`
- `share_plus`

Added/modified some tests to run on Linux

Skipped `sensor_plus`

## Related Issues

Closes #1104 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: .github/workflows/scripts/integration-test.sh [hunk 1]
```diff
@@ -14,6 +14,8 @@ fi
 if [ "$ACTION" == "linux" ]
 then
   sudo apt-get install ninja-build libgtk-3-dev
+  export DISPLAY=:99
+  sudo Xvfb -ac :99 -screen 0 1280x1024x24 > /dev/null 2>&1 &
```
Comment (abd99) [inline .github/workflows/scripts/integration-test.sh:18]: Had to add a virtual framebuffer to enable UI operations on a headless CI

File: .github/workflows/connectivity_plus.yaml [hunk 2]
```diff
@@ -133,6 +133,21 @@ jobs:
...
+    timeout-minutes: 30
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v2
+      - name: "Install Flutter"
+        run: ./.github/workflows/scripts/install-flutter.sh stable
+      - name: "Install Tools"
+        run: ./.github/workflows/scripts/install-tools.sh
+      - name: "Install NetworkManager"
+        run: sudo apt install network-manager
```
Comment (abd99) [inline .github/workflows/connectivity_plus.yaml:147]: Installed NetworkManager manually to help detect connectivity status on the CI

File: packages/battery_plus/battery_plus/example/integration_test/battery_plus_test.dart [hunk 3]
```diff
@@ -13,11 +13,8 @@ import 'package:integration_test/integration_test.dart';
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
 
-  final bool batteryLevelIsImplemented = Platform.isAndroid || Platform.isMacOS;
-  final bool batteryStateIsImplemented = Platform.isAndroid ||
-      Platform.isIOS ||
-      Platform.isMacOS ||
-      Platform.isWindows;
```
Comment (miquelbeltran) [inline packages/battery_plus/battery_plus/example/integration_test/battery_plus_test.dart:20]: Why of this change?

Comment (abd99) [reply packages/battery_plus/battery_plus/example/integration_test/battery_plus_test.dart]: I thought those tests were already running everywhere else. Didn't know that they failed on simulators on the CI. Changed to a working version.

**Comment by miquelbeltran** [comment]
Just a heads-up, there is another PR covering the same functionality here https://github.com/fluttercommunity/plus_plugins/pull/1137 which was opened earlier.

**Comment by abd99** [comment]
Yup, I saw that it had failing tests and I had already done a lot of work on mine. So, I opened a new one with my work PR anyway.

**Comment by miquelbeltran** [comment]
Sorry @abd99 we went with the other PR that was opened earlier, as both were similar.

**Comment by abd99** [comment]
No problem. Great work, @tibi77!

**Pull request closed without merging**